__tests__/test.js

```diff
@@ -32,4 +32,21 @@ describe('Savitzky–Golay test', function() {
     }
     expect(ans[0]).toBe(0);
   });
+
+  it('Second derivative test', function() {
+    let options = {
+      windowSize: 7,
+      derivative: 2,
+      polynomial: 3,
+    };
+    let data = new Array(100);
+    for (let i = 0; i < data.length; i++) {
+      data[i] = Math.pow(i, 3) - 2 * Math.pow(i, 2);
+    }
+    let ans = savitzkyGolay(data, 1, options);
+    expect(ans).toHaveLength(94);
+    for (let j = 0; j < ans.length; j++) {
+      expect(ans[j]).toBeCloseTo(6 * (j + 3) - 4, 2);
```

Comment:
The second derivative of `x^3 - 2*x^2` is `6*x - 4`, so I don't understand where do you calculate the `(j+3)`?

Reply:
Because the `windowSize` used for this test is 7 and there is no padding, `ans` corresponds to positions 3-97 of `data`. This is where the `+3` comes from. 

There is similar to the first test (`Savitzky–Golay test`): the `windowSize` is 5, and so we have `j + 2` in the check:
```
expect(ans[j]).toBeCloseTo(data[j + 2], 0.5)
```

Reply:
Sorry, you are right, LGTM